lib/web_game.rb

```diff
@@ -33,18 +36,27 @@ def save_game(game_id)
     store(data)
   end
 
-  def start_game(game_id)
+  def start_game(game_id = @uuid.generate)
     data = { game_id => nil }
     store(data)
+    game_id
   end
 
   def get_game(game_id)
     @datastore.load_game(game_id)
   end
 
-  def check_win(game_id, player)
+  def check_win(game_id, _current_player)
+    players = [[1, Player.create_player(1)], [2, Player.create_player(2)]]
     load_state(game_id)
-    @game.check_win(player.get_mark)
+    players.each do |player|
+      return player[0] if player[1] && @game.check_win(player[1].get_mark)
+    end
+    false
+  end
+
+  def game_end?(game_id, player)
+    check_win(game_id, player) || draw?(game_id)
   end
```

Comment:
_current_player is no longer being used in this method.
Does it have to be passed as a parameter?

Also, this method now violates SRP:
 * it creates players
 * it also checks for win

This method also returns 2 different types:
 * a boolean on line 55
 * a Player on line 53
Is that intended? This can lead to unintended results `bug`

Reply:
oh shoot.. I forgot to refactor that.Thanks for noticing it. let me take care of it

Reply:
@Opio-Emmanuel-Omona